phosphor-auditlog/alog-parser.cpp

```diff
@@ -0,0 +1,427 @@
+    std::error_code ec;
+
+    /* Expect the file has already been created */
+    if (!std::filesystem::exists(filePath, ec))
+    {
+        lg2::error("File {FILE} doesn't already exist.", "FILE", filePath);
+        return false;
+    }
+
+    // Create/Open file using truncate
+    parsedStream.open(filePath, std::ios::out);
+    if (parsedStream.fail())
+    {
+        lg2::error("Failed to open {FILE}", "FILE", filePath);
+        throw sdbusplus::xyz::openbmc_project::Common::File::Error::Open();
+        return false;
+    }
+
+    return true;
+}
+
+void ALParser::processEvents()
+{
+    // Loop over all the events
+    while (auditNextEvent())
+    {
+        parseEvent();
+    }
+}
+
+void ALParser::doParse()
+{
+    lg2::debug("Parsing All");
+    processEvents();
+}
+
+void ALParseLatest::parseRecord()
+{
+    nlohmann::json parsedEntry;
+
+    if (formatEntry(parsedEntry))
+    {
+        // Keep list limited to maxLeftCount
+        if (parsedEntries.size() >= maxLeftCount)
+        {
+            parsedEntries.pop_back();
+        }
+
+        parsedEntries.emplace_front(parsedEntry.dump());
+    }
+
+    return;
+}
+
+size_t ALParseLatest::writeParsedEntries()
+{
+    /* Add newest events to the file */
+    for (const auto& iter : parsedEntries)
+    {
+        parsedStream << iter << '\n';
+    }
+
+    auto parsedCount = parsedEntries.size();
+    parsedEntries.clear();
+
+    lg2::debug("maxLeftCount: {MAXCOUNT} parsedCount: {PARSED}", "MAXCOUNT",
+               maxLeftCount, "PARSED", parsedCount);
+
+    return parsedCount;
+}
+
+void ALParseLatest::doParse()
+{
+    lg2::debug("Parsing maxCount: {MAXCOUNT}", "MAXCOUNT", maxCount);
+    processEvents();
+
+    auto parsedCount = writeParsedEntries();
+
+    // Process next file if needed to reach desired # entries
+    while (parsedCount < maxLeftCount)
+    {
+        maxLeftCount -= parsedCount;
+
+        if (!initNextLog())
+        {
+            // No more files to parse
+            break;
+        }
+
+        processEvents();
+
+        /* Add newest events to the file */
+        parsedCount = writeParsedEntries();
+    }
+}
+
+bool ALParseLatest::initNextLog()
+{
+    if (au != nullptr)
+    {
+        lg2::debug("initNextLog: destroying existing au");
+        auparse_destroy(au);
+        au = nullptr;
+    }
+
+    /* Determine path of next log file to process.
+     * Newest file has no extension and matches 0 index value.
+     */
+    std::string logFilePath = "/var/log/audit/audit.log";
+
+    if (logFileIdx)
+    {
+        logFilePath = std::format("/var/log/audit/audit.log.{}",
+                                  std::to_string(logFileIdx));
+    }
+    else
+    {
+        logFilePath = "/var/log/audit/audit.log";
+    }
+
+    lg2::debug("initNextLog: Initialize for {FILE}", "FILE", logFilePath);
+    au = auparse_init(AUSOURCE_FILE, logFilePath.c_str());
+
+    if (au != nullptr)
+    {
+        logFileIdx++;
+    }
+
+    return (au != nullptr);
```

Comment:
Perhaps this can be clearer like this?

```
    if (au != nullptr)
    {
        logFileIdx++;
        return true;
    }
    return false;
```

Reply:
Agree, I've changed it.